samples/storageRetryConditionallyIdempotent.js

```diff
@@ -0,0 +1,42 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * This application demonstrates how to perform basic operations on buckets with
+ * the Google Cloud Storage API.
+ *
+ * For more information, see the README.md under /storage and the documentation
+ * at https://cloud.google.com/storage/docs.
+ */
+
+function main() {
+  // [START storage_retry_conditionally_idempotent]
+  /**
+   * TODO(developer): Uncomment the following lines before running the sample.
+   */
+  // Imports the Google Cloud client library and idempotency strategy options
+  const {Storage, IdempotencyStrategy} = require('@google-cloud/storage');
+
+  // Creates a client
+  const storage = new Storage({
+    retryOptions: {
```

Comment:
I feel like this could use a bit more in-line commentary since the configs will be specific to the language.